### DISCUSSION TITLE
best --player-external-http --player-external-http-port 2

### DISCUSSION THREAD

**Discussion by HalowTV** [streamlink/streamlink#5643] [Q&A] [answered]
### Description

hey i try for what mp4 please someone can help me please
This is how it works but not like mp4
nohup streamlink -p mpv "https://ok.ru/video/6452461242937" best --player-external-http --player-external-http-port 2. &

working so
http://127.0.0.1:2/index.m3u8
but so is not working 
http://127.0.0.1:2.mp4

**Comment by bastimeyer** [comment, marked as answer]
There is so much wrong here...

First, `--player`/`-p` is mutually exclusive to `--player-external-http`, which means `--player` is useless here, and second, `2` is a terrible choice for the HTTP server's port. You should use a TCP port above 1000 that's unassigned.

Third, Streamlink's output on its HTTP server is a simple progressive stream on the `/` path. There is no HLS proxying or passthrough here, so `/index.m3u8` is incorrect.

And finally, `http://127.0.0.1:2.mp4` doesn't make any sense. It's not a valid URL.

If you're looking for remuxing the output into an mp4 container, then you need to pipe the output using `--stdout` into ffmpeg. This has been answered here lots of times already.

**Comment by HalowTV** [comment]
thx mate
